lessons/python_basics/practice_code.py

```diff
@@ -0,0 +1,277 @@
+
+#We're working within an interpreter (similar to Matlab),
+#so we type code directly into the interpreter
+
+print "Hello World"
+
+
+########## DATA TYPES: NUMBERS & STRINGS ##########
+
+#Python has five standard data types: numbers, string, lists,
+# tuples, and dictionaries.  Let's deal with numbers and 
+#strings first.  Numbers can be ints (and longs in older 
+#versions), floats or complex.
+
+#One advantage of the interpreter is we can solve math
+#problems directly in it
+
+2 + 2
+12.*3				#The int gets typecast to float
+(1.+1j)/2.**0.5	
+1/2					#Be warned - this returns zero in Python 2.x!
+
+#Note that running this .py as a script will not print those
+#numbers to screen, but running the commands directly on the
+#command line will.
+
+#To figure out the data type of a variable, we can use:
+
+x = 47
+print type(x)
+x = "spam"
+print type(x)
+
+#Variables don't have pre-declared types in Python, so you can
+#assign any data type to any variable.
+
+#As an aside, to get documentation on most things within Python:
+
+help(type)
+
+#We can also use the magic function "type?" in IPython.
+
+#Print is a intelligent, and we can pass it all sorts of weird 
+#things.
+
+x = 42
+print "This is a string, and these are numbers: ", x, 324.13, ", and another string."
+
+#String are character arrays, and their operations are an art form.
+#Here is a smattering of commands.
+
+s = "This is a test string"
+print s
+print s[0]			#Array syntax in python - to access element i of an array, we do array[i]
+print s[-3]			#Negative indicies access elements backwards from the end of the list
+print s[2:4]		#Accesses elements 2 - 3
+print s[6:]			#Accesses elements 6 and above
+print s + ' ' + s	#Strings can be concatenated
+print s*10			#Or multiplied...
+
+print s.replace('a test','the bestest') # print a string with a replaced word
+
+#Above, replace is a method, a function that is a part of an 
+#object (in this case, s).  Tab-completion can be used to determine 
+#what methods exist for a given object.
+
+#Perform equality tests on strings; the output is a boolean.
+#(Python doesn't care about using ' or " for strings.)
+
+print s == 'This is a test string'
+
+
+
+########## DATA TYPES: LISTS, TUPLES & DICTS ##########
+
+#Python has two array-like things.  The first is called a "list", 
+#which can hold any data types:
+
+mylist = [1, "2", 3, 4.000, "five"]
+print len(mylist)						#Prints number of elements in list
+print mylist[2:]						#Indexing is the same as for strings
+print mylist + mylist					#Adding lists combines them
+print mylist.index(4.)					#Return the index of a list element
+
+#The second is called a "tuple", which is an immutable list (nothing 
+#can be added or subtracted) whose elements also can't be reassigned.  
+#They're easier on memory and processor than lists.
+
+mytuple = (1, "2", 3j, 4.000, "five")
+
+#Python also has dictionaries, which are unordered sets of values accessible 
+#by string keys.
```

Comment:
The key can be anything that's "hashable", including numbers, or custom classes.